wallit/signsend.go

```diff
@@ -21,10 +21,201 @@ import (
+			if txo == nil || txo.PkScript == nil || txo.Value == 0 {
+				return nil, fmt.Errorf("BuildAndSign arg invalid txo")
+			}
+			tx.AddTxOut(txo)
+		}
+		// add all the txins, first refenecing the prev outPoints
+		for i, u := range utxos {
+			tx.AddTxIn(wire.NewTxIn(&u.Op, nil, nil))
+			// set sequence field if it's in the portxo
+			if w.Rbf == true {
+				u.Seq = 2
+			} else {
+				u.Seq = 1
+			}
+			tx.TxIn[i].Sequence = 4294967295 - u.Seq
+		}
+		// sort txouts in place before signing.  txins are already sorted from above
+		txsort.InPlaceSort(tx)
+
+		if changeOut != nil {
+			txos = append(txos, changeOut)
+		}
+
+		// call a aprt of buildandsign over here
+		if len(utxos) == 0 || len(txos) == 0 {
+			return nil, fmt.Errorf("BuildAndSign args no utxos or txos")
+		}
+		// sort input utxos first.
+		sort.Sort(portxo.TxoSliceByBip69(utxos))
+
+		padding += 10
+		feePerByte += 20 // steps of 20 or 10??
+		// RerallySend the transaction
+		hCache := txscript.NewTxSigHashes(tx)
+		// make the stashes for signatures / witnesses
+		sigStash := make([][]byte, len(utxos))
+		witStash := make([][][]byte, len(utxos))
+
+		for i, _ := range tx.TxIn {
+			// get key
+			priv := w.PathPrivkey(utxos[i].KeyGen)
+			log.Printf("signing with privkey pub %x\n", priv.PubKey().SerializeCompressed())
+
+			if priv == nil {
+				return nil, fmt.Errorf("SendCoins: nil privkey")
+			}
+
+			// sign into stash.  3 possibilities:  legacy PKH, WPKH, WSH
+			if utxos[i].Mode == portxo.TxoP2PKHComp { // legacy PKH
+				sigStash[i], err = txscript.SignatureScript(tx, i,
+					utxos[i].PkScript, txscript.SigHashAll, priv, true)
+				if err != nil {
+					return nil, err
+				}
+			}
+			if utxos[i].Mode == portxo.TxoP2WPKHComp { // witness PKH
+				witStash[i], err = txscript.WitnessScript(tx, hCache, i,
+					utxos[i].Value, utxos[i].PkScript, txscript.SigHashAll, priv, true)
+				if err != nil {
+					return nil, err
+				}
+				//}
+			}
+			if utxos[i].Mode == portxo.TxoP2WSHComp { // witness script hash
+				sig, err := txscript.RawTxInWitnessSignature(tx, hCache, i,
+					utxos[i].Value, utxos[i].PkScript, txscript.SigHashAll, priv)
+				if err != nil {
+					return nil, err
+				}
+				// witness stack has the signature, items, then the previous full script
+				witStash[i] = make([][]byte, 2+len(utxos[i].PreSigStack))
+
+				// sig comes first (pushed to stack last)
+				witStash[i][0] = sig
+
+				// after stack comes PostSigStack items
+				for j, element := range utxos[i].PreSigStack {
+					witStash[i][j+1] = element
+				}
+
+				// last stack item is the pkscript
+				witStash[i][len(witStash[i])-1] = utxos[i].PkScript
+			}
+
+		}
+		// swap sigs into sigScripts in txins
+		for i, txin := range tx.TxIn {
+			if sigStash[i] != nil {
+				txin.SignatureScript = sigStash[i]
+			}
+			if witStash[i] != nil {
+				txin.Witness = witStash[i]
+				txin.SignatureScript = nil
+			}
+		}
+
+		log.Printf("tx: %s", TxToString(tx))
+		w.NewOutgoingTx(tx)
```

Comment:
This won't work; it will just try to broadcast txs that have locktimes in the future, which will be rejected.